tests/integration/pilot/analysis/analysis_test.go

```diff
@@ -102,6 +102,145 @@ spec:
 			retry.UntilSuccessOrFail(t, func() error {
 				return expectVirtualServiceStatus(t, ns, false)
 			})
+			// Apply bad config (ServiceEntry missing address)
+			t.ConfigIstio().YAML(ns.Name(), `
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: vm-1
+spec:
+  hosts:
+  - dummy.github.com
+  addresses:
+  - 1.1.1.1
+  location: MESH_EXTERNAL
+  ports:
+  - name: tcp
+    number: 22
+	protocol: TCP
+`).ApplyOrFail(t)
+			// Status should report error
+			retry.UntilSuccessOrFail(t, func() error {
+				return expectServiceEntryStatus(t, ns, nil, true)
+			}, retry.Timeout(time.Second*5))
+		})
+}
+
+func TestServiceEntryUpdatesStatus(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(t framework.TestContext) {
+			ns := namespace.NewOrFail(t, namespace.Config{
+				Prefix:   "default",
+				Inject:   true,
+				Revision: "",
+				Labels:   nil,
+			})
+
+			// create ServiceEntry
+			t.ConfigIstio().YAML(ns.Name(), `
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: vm-1
+spec:
+  hosts:
+  - dummy.github.com
+  addresses:
+  - 1.1.1.1
+  location: MESH_EXTERNAL
+  ports:
+  - name: tcp
+    number: 22
+    protocol: TCP
+`).ApplyOrFail(t)
+			retry.UntilSuccessOrFail(t, func() error {
+				// we should expect an empty array not nil
+				return expectServiceEntryStatus(t, ns, nil, false)
```

Comment:
we do expect nil here though by the look, is the comment just old?

Reply:
yeah it's an old comment from copying the seemingly similar workload entry test